component/newstorage/formattedstore/formattedstore.go

```diff
@@ -20,25 +21,27 @@ const (
 	expressionTagNameOnlyLength     = 1
 	expressionTagNameAndValueLength = 2
 
-	failFormat               = `failed to format %s "%s": %w`
-	failFormatTags           = "failed to format tags: %w"
-	failDeformat             = `failed to deformat %s "%s" returned from the underlying store: %w`
-	failQueryUnderlyingStore = "failed to query underlying store: %w"
+	failFormat                     = `failed to format %s "%s": %w`
+	failFormatData                 = `failed to format data: %w`
+	failDeformat                   = `failed to deformat %s "%s" returned from the underlying store: %w`
+	failQueryUnderlyingStore       = "failed to query underlying store: %w"
+	failGetValueUnderlyingIterator = "failed to get formatted value from the underlying iterator: %w"
 )
 
-var logger = log.New("formattedstore")
+var (
+	logger = log.New("formattedstore")
 
-var errInvalidQueryExpressionFormat = errors.New("invalid expression format. " +
-	"it must be in the following format: TagName:TagValue")
+	errEmptyKey                     = errors.New("key cannot be empty")
+	errInvalidQueryExpressionFormat = errors.New("invalid expression format. " +
+		"it must be in the following format: TagName:TagValue")
+)
 
 // Formatter represents a type that can convert data between two formats.
 type Formatter interface {
```

Comment:
I had to change the Formatter interface in order to accommodate the EDV Encrypted Formatter, which is more "restrictive" due to the fact that key and tag formatting is not reversible (as they use hashing functions). You'll find that some of the format provider functions have been changed to accommodate this. I left some comments in the code explaining why certain things here work the way they do. The whole idea here is to allow someone to use the Encrypted Formatted with any store, not just the EDV REST provider.. You'll find unit tests demonstrating the use of an Encrypted Formatter + mem-store combo.

(while you could theoretically pass the EDV REST provider in as the underlying store to a formatted provider, there's no reason to, as the REST provider already requires the encrypted formatter... so you would end up with double formatting/deformatting operations)